src/tests/integration-test

```diff
@@ -1376,6 +1384,74 @@ class Luks(UDisksTestCase):
+        encrypted = self.setup_crypto_device()
+        # wrong password, has bytes after a trailing NUL
+        self.assertRaises(GLib.GError, encrypted.call_unlock_sync,
+                          '', self.keyfile_options(b's3kr1t\0X'), None)
+        self.assertRaises(GLib.GError, encrypted.call_unlock_sync,
+                          '', self.keyfile_options(b's3kr1t\n'), None)
+        # correct password, specified as keyfile
+        encrypted.call_unlock_sync('', self.keyfile_options(b's3kr1t'), None)
+        encrypted.call_lock_sync(no_options, None)
+
+    def test_plaintext_keyfile(self):
+        '''Setup a device using a plaintext keyfile.'''
+        # Using a plaintext keyfile should be equivalent to passphrase
+        self.fs_create(None, 'ext4', GLib.Variant('a{sv}', {
+            'encrypt.passphrase': GLib.Variant('ay', b's3kr1t'),
+            'label': GLib.Variant('s', 'treasure')}))
+
+        crypt_obj = self.client.get_object(self.udisks_block().get_object_path())
+        encrypted = crypt_obj.get_property('encrypted')
+        encrypted.call_lock_sync(no_options, None)
+
+        # wrong password
+        self.assertRaises(GLib.GError, encrypted.call_unlock_sync,
+                          'h4ckpassword', no_options, None)
+        self.assertRaises(GLib.GError, encrypted.call_unlock_sync,
+                          '', self.keyfile_options(b's3kr1t\0X'), None)
+        self.assertRaises(GLib.GError, encrypted.call_unlock_sync,
+                          '', self.keyfile_options(b's3kr1t\n'), None)
+
+        # correct password
+        encrypted.call_unlock_sync('', self.keyfile_options(b's3kr1t'), None)
+        encrypted.call_lock_sync(no_options, None)
+
+        # correct password, specified as passphrase
+        encrypted.call_unlock_sync('s3kr1t', no_options, None)
+        encrypted.call_lock_sync(no_options, None)
+
+    def test_binary_keyfile(self):
+
+        KEYFILE = b's\0me \bina\ry \xda\ta\0\0\1\1\xff\xff'
+
+        self.fs_create(None, 'ext4', GLib.Variant('a{sv}', {
+            'encrypt.passphrase': GLib.Variant('ay', KEYFILE),
+            'label': GLib.Variant('s', 'treasure')}))
+        crypt_obj = self.client.get_object(self.udisks_block().get_object_path())
+        encrypted = crypt_obj.get_property('encrypted')
+        encrypted.call_lock_sync(no_options, None)
+
+        # wrong password
+        missing_byte = self.keyfile_options(KEYFILE[:-1])
+        extra_bytes = self.keyfile_options(KEYFILE+b'\0X')
+        self.assertRaises(GLib.GError, encrypted.call_unlock_sync,
+                          'h4ckpassword', no_options, None)
+        self.assertRaises(GLib.GError, encrypted.call_unlock_sync,
+                          '', missing_byte, None)
+        self.assertRaises(GLib.GError, encrypted.call_unlock_sync,
+                          '', extra_bytes, None)
+
+        # correct password
+        encrypted.call_unlock_sync('', self.keyfile_options(KEYFILE), None)
+
```

Comment:
Could you please also add a test with ASCII/text data containing ``'\n'``? Thanks!